src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/TraceLocation.swift

```diff
@@ -108,5 +111,41 @@ struct TraceLocation {
 		let encodedByteRepresentation = byteRepresentation.base32EncodedString
 		return String(format: "https://e.coronawarn.app/c1/%@", encodedByteRepresentation).uppercased()
 	}
+}
+
+extension TraceLocation {
+	init?(qrCodeString: String) {
+		
+		guard let data = qrCodeString.base32DecodedData else {
```

Comment:
please move it inside the struct TraceLocation and use the usual marks

Reply:
We need the extension as we want to keep the normal init valid and to initialize the struct with the normal properties, i added the MARK